src/test/java/stepdefinition/MainPageStep.java

```diff
@@ -0,0 +1,79 @@
+package stepdefinition;
+
+import com.giantshoulder.pageobject.MainPage;
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.When;
+
+/**
+ * @author Gabriella_Havadtoi
+ */
+
+public class MainPageStep extends BaseStepDefinition {
+
+    MainPage mainPage = new MainPage(driver);
+
+    @Given("I navigate to the main page")
+    public void i_navigate_to_main_page() {
+        driver.navigate().to(BASE_URL);
+    }
+
+    @And("I accept the data handling dialog box")
+    public void i_accept_the_data_handling_dialog_box() {
+        mainPage.acceptDataConsentPanel();
+    }
+
+    @And("I click on the Subscribe button")
+    public void i_click_on_the_subscribe_button() {
+        mainPage.clickOnSubscribeButton();
+    }
+
+    @And("I chose 'Learn faster' newsletter")
+    public void i_chose_learn_faster_newsletter_types() {
+        mainPage.clickOnNewsLetterButtonFor("learn faster");
```

Comment:
If you are adamant on the switch approach, then these "I chose..." steps can be merged into one.

Reply:
Discussed offline: instead of switch separate scenarios should be used for each subscription option.

Reply:
Fixed in 5e8e0dc07a1b347edd3a35b7fc060122e44581be!